microsoft/testsuites/core/azure_image_standard.py

```diff
@@ -582,10 +582,143 @@ def verify_serial_console_is_enabled(self, node: Node) -> None:
         use_new_environment=True,
     )
     def verify_bash_history_is_empty(self, node: Node) -> None:
-        cat = node.tools[Cat]
-        path_bash_history = node.get_pure_path("/root/.bash_history")
-        if path_bash_history:
-            bash_history = cat.read(str(path_bash_history), sudo=True)
+        path_bash_history = "/root/.bash_history"
+        cmd_result = node.execute(f"ls -lt {path_bash_history}", sudo=True, shell=True)
+        if 0 == cmd_result.exit_code:
+            cat = node.tools[Cat]
+            bash_history = cat.read(path_bash_history, sudo=True)
             assert_that(bash_history).described_as(
                 "/root/.bash_history is not empty, this image is not prepared well."
             ).is_empty()
+
+    @TestCaseMetadata(
+        description="""
+        This test will check error, failure, warning messages from demsg,
+         /var/log/syslog or /var/log/messages file.
+
+        Steps:
+        1. Get failure, error, warning messages from dmesg, /var/log/syslog or
+         /var/log/messages file.
+        2. If any unexpected failure, error, warning messages excluding ignorable ones
+         existing, fail the case.
+        """,
+        priority=1,
+    )
+    def verify_boot_error_fail_warnings(self, node: Node) -> None:
+        # pattern to get failure, error, warnings from dmesg, syslog/messages
+        error_fail_warnings_pattern: List[Pattern[str]] = [
+            re.compile(r"^(.*fail.*)$", re.MULTILINE),
+            re.compile(r"^(.*error.*)$", re.MULTILINE),
+            re.compile(r"^(.*warning.*)$", re.MULTILINE),
+        ]
+
+        # ignorable failure, error, warnings got confirmed
+        error_fail_warnings_ignorable_str_list: List[str] = [
```

Comment:
Move them to class level, and use `re.compile` to precompile them.

Reply:
updated